src/demo/java/demo/customplugins/ExecutionTimeConnectionPlugin.java

```diff
@@ -0,0 +1,118 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package demo.customplugins;
+
+import com.mysql.cj.jdbc.ha.ca.plugins.IConnectionPlugin;
+import com.mysql.cj.log.Log;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This connection plugin tracks the execution time of all methods executed with
+ * enhanced instance monitoring enabled.
+ * During the cleanup phase when {@link ExecutionTimeConnectionPlugin#releaseResources()}
+ * is called, this plugin logs all the methods executed and time spent on each execution
+ * in milliseconds.
+ */
+public class ExecutionTimeConnectionPlugin implements IConnectionPlugin {
+  final long initializeTime;
+  final IConnectionPlugin nextPlugin;
+  private final Log logger;
+  private final Map<String, Long> results = new HashMap<>();
+
+  public ExecutionTimeConnectionPlugin(
+      IConnectionPlugin nextPlugin,
+      Log logger) {
+    this.nextPlugin = nextPlugin;
+    this.logger = logger;
+
+    initializeTime = System.currentTimeMillis();
+  }
+
+  @Override
+  public Object execute(
+      Class<?> methodInvokeOn,
+      String methodName,
+      Callable<?> executeSqlFunc)
+      throws Exception {
+    // This `execute` measures the time it takes for the remaining connection plugins to
+    // execute the given method call.
+    // In this sample, the next connection plugin is the `NodeMonitoringConnectionPlugin`,
+    // so this measurement includes the time it takes for the
+    // `NodeMonitoringConnectionPlugin` to finish all its necessary setups.
+    final long startTime = System.nanoTime();
+    final Object result =
+        this.nextPlugin.execute(methodInvokeOn, methodName, executeSqlFunc);
+    final long elapsedTime = System.nanoTime() - startTime;
+    results.merge(
+        methodName,
+        elapsedTime / 1000000,
+        Long::sum);
+
+    return result;
+  }
+
+  @Override
+  public void releaseResources() {
+    // Output the aggregated information from all methods called throughout the lifespan
+    // of the current connection.
+    final long elapsedTime = System.nanoTime() - initializeTime;
+    final String leftAlignFormat = "| %-19s | %-10s |\n";
+    final StringBuilder logMessage = new StringBuilder();
+
+    logMessage.append("** ExecutionTimeConnectionPlugin Summary **\n");
+    logMessage.append(String.format(
+        "Plugin Uptime: %ds\n",
+        elapsedTime / 1000000
+    ));
+
+    logMessage
+        .append("** Method Execution Time With Enhanced Instance Monitoring **\n")
```

Comment:
I'm not sure I understand what "Enhanced Instance Monitoring" means in this context. It's just a sample plugin that has no relation to introspection.